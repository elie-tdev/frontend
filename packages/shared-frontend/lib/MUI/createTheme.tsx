import { createTheme as createMuiTheme } from '@mui/material/styles'

import { makeDefaultTheme } from './defaultTheme'
import { BAIThemeOptions } from './types'
import { createDefaultPaletteOptions } from './utils/palette'
import { createDefaultSpacingOptions } from './utils/spacing'

const createThemeBase = (themeOptions: BAIThemeOptions, isTest = false) => {
  createDefaultSpacingOptions(themeOptions)
  createDefaultPaletteOptions(themeOptions)
  return makeDefaultTheme(createMuiTheme(themeOptions), !isTest)
}

export const createTheme = (themeOptions: BAIThemeOptions = {}) =>
  createThemeBase(themeOptions)

export const createThemeTest = (themeOptions: BAIThemeOptions = {}) =>
  createThemeBase(themeOptions, true)
