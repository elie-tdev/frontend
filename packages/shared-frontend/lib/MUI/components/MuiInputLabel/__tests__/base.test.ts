import { MuiInputLabel } from '../'
import { createThemeTest } from '../../../createTheme'

describe('Input Label', () => {
  const theme = createThemeTest()
  const component = MuiInputLabel(theme)

  it('should be consistent with previous versions: ', () => {
    expect(component).toMatchSnapshot()
  })
})
