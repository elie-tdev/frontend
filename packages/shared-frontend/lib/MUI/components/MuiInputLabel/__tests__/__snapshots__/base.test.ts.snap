// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Label should be consistent with previous versions:  1`] = `
Object {
  "styleOverrides": Object {
    "filled": Object {
      "&:not(.MuiInputLabel-shrink):not(.MuiInputLabel-sizeSmall)": Object {
        "transform": "translate(12px, 18px) scale(1)",
      },
    },
  },
}
`;
